Musicca/app/src/main/java/com/example/musicca/fragments/MultiFilterDialogFragment.java

```diff
@@ -0,0 +1,106 @@
+
+    private static final String TITLE_ARG = "title";
+
+    private String YEAR_SELECTED;
+    private TextView yearReleased;
+    private Spinner yearSpinner;
+    private TextView tvSongTitle;
+    private EditText etSongTitle;
+    private TextView tvSongArtist;
+    private EditText etSongArtist;
+    private Button btnApplyFilter;
+
+    public MultiFilterDialogFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+        YEAR_SELECTED = adapterView.getSelectedItem().toString();
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> adapterView) {
+
+    }
+
+    // Defines the listener interface with a method passing back data result.
+    public interface onFilterActionListener{
+        void onFinishFilterDialog(String year, String songTitle, String songArtist);
+    }
+
+    public static MultiFilterDialogFragment newInstance(String title) {
+        MultiFilterDialogFragment frag = new MultiFilterDialogFragment();
+        Bundle args = new Bundle();
+        args.putString(TITLE_ARG, title);
+        frag.setArguments(args);
+        return frag;
+    }
+
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_multi_filter_dialog, container);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        yearReleased = view.findViewById(R.id.yearReleased);
+        yearSpinner = view.findViewById(R.id.yearSpinner);
+        tvSongTitle = view.findViewById(R.id.tvSongTitle);
+        etSongTitle = view.findViewById(R.id.etSongTitle);
+        tvSongArtist = view.findViewById(R.id.tvSongArtist);
+        etSongArtist = view.findViewById(R.id.etSongArtist);
+        btnApplyFilter = view.findViewById(R.id.btnApplyFilter);
+
+        // Fetch arguments from bundle and set title
+        String title = getArguments().getString(TITLE_ARG, "Apply Filter");
+        getDialog().setTitle(title);
+        getDialog().getWindow().setSoftInputMode(
+                WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        yearSpinner.setOnItemSelectedListener(this);
+
+        btnApplyFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                onFilterActionListener listener = (onFilterActionListener) getActivity();
+                listener.onFinishFilterDialog(YEAR_SELECTED,
+                        etSongTitle.getText().toString().toLowerCase().trim(),
+                        etSongArtist.getText().toString().toLowerCase().trim());
```

Comment:
I like the thinking here to lower case and trim the strings, to avoid different cases resulting in mismatch in comparison!